python/ray/dag/compiled_dag_node.py

```diff
@@ -52,22 +52,30 @@ def do_exec_compiled_task(
     self._dag_cancelled = False
 
     try:
-        self._input_channels = [i for i in inputs if isinstance(i, Channel)]
+        self._input_channels = list(
+            set(filter(lambda x: isinstance(x, Channel), inputs[0]))
```

Comment:
Nit: it's a bit more pythonic/readable to do list comprehension for these cases.

Reply:
refactored